lib/compat/wordpress-6.0/class-wp-webfonts.php

```diff
@@ -291,4 +300,24 @@ public function generate_styles() {
 
 		return $styles;
 	}
+
+	/**
+	 * Preload webfonts to improve performance
+	 *
+	 * @return void
+	 */
+	public function preload_webfonts() {
+		foreach ( $this->get_fonts() as $font ) {
+			if ( ! empty( $font['preload'] ) && $font['preload'] ) {
+				foreach( $font['src'] as $src ) {
+					$src_as_array = explode( '.', $src );
+					$file_type    = end( $src_as_array );
```

Comment:
Micro-optimization suggestion:
```suggestion
					$file_type_pos = strrpos( $src, '.' ) + 1;
					$file_type     = substr( $src, $file_type_pos );
```

Why? `explode` creates an array which consumes a bit more memory and time. To see it in action, here's a benchmark running on PHP 7.4 over 1 million cycles https://3v4l.org/EnOB1#v7.4.28.

Results:
| Implementation | Average μs |
| ------- | ------- |
| `explode` and `end` | 0.2440 μs |
| `strrpos` and `substr` | 0.0959 μs |

Reply:
BTW This code is not needed if the preload functionality instead grabs the font srcs from the provider's `WP_Webfonts_Provider_Local::order_src()` as [suggested here](https://github.com/WordPress/gutenberg/pull/39391/files#r826658071).